system/lib/libc/musl/src/stdio/fflush.c

```diff
@@ -34,11 +34,22 @@ int fflush(FILE *f)
 	}
 
 	/* If reading, sync position, per POSIX */
+#if __EMSCRIPTEN__
+	/* Handle failues of lseek, which can happen in emscripten, e.g. for stdin etc */
+	if (f->rpos != f->rend) {
+		if (f->seek(f, f->rpos-f->rend, SEEK_CUR) == 0) {
```

Comment:
If fseek fails shouldn't flush return -1 to indicate failure? Or is this seek optional for some reason?

Reply:
I guess if we're flushing stdout we don't want it to fail because it's not seekable

Reply:
This seeks I believe is kind of optional.  What is trying to do is that read-ahead buffer should be discarded and the FD should be seek'd back to the current position.   On linux you can seek backward in stdin, but not in emscripten.

We should try to fix emscripten's stdin to make it seekable, but that seem harder than making this small change to fflush.